info: add initial implementation

ee7 opened the pull request:
This PR adds the `configlet info` command, which aims to provide information that helps the maintainers of Exercism tracks.

The initial implementation prints the following information for a track:
- Every concept that is not a prerequisite for any Practice Exercise
- Every concept that is not practiced by any Practice Exercise
- The difference between those two lists: concepts that are a prerequisite, but aren't practiced
- Exercises from `exercism/problem-specifications` that are neither implemented nor in `foregone`
- The number of exercises and concepts

In the future, the command will likely have various options to control what output is shown.

Discussion: https://github.com/exercism/configlet/issues/180

---

Edit: below output updated on `2021-09-30`

```
$ ./configlet -t elixir info
Concepts that aren't a prerequisite for any Practice Exercise:
access-behaviour
alias
docs
import
io
links
try-rescue
try-rescue-else-after
typespecs

Concepts that aren't practiced by any Practice Exercise:
access-behaviour
alias
docs
dynamic-dispatch
import
io
links
streams
try-rescue
try-rescue-else-after
typespecs

Concepts that are a prerequisite, but aren't practiced by any Practice Exercise:
dynamic-dispatch
streams

There are 11 exercises from `exercism/problem-specifications` that are neither implemented nor in the track config `exercises.foregone` array:
counter
error-handling
hangman
ledger
lens-person
nucleotide-codons
octal
paasio
point-mutations
tree-building
trinary

Track summary:
 35 Concept Exercises (plus 1 work-in-progress)
116 Practice Exercises (plus 0 work-in-progress)
151 Exercises in total (plus 1 work-in-progress)
 54 Concepts
```

ErikSchierboom commented on src/info/info.nim:
```diff
@@ -0,0 +1,72 @@
+    result.incl item["slug"].getStr()
+
+proc getPrereqs(j: JsonNode): HashSet[string] =
+  ## Returns the dedeuplicated values of every practice exercise `prerequisites` key.
```
```suggestion
  ## Returns the deduplicated values of every practice exercise `prerequisites` key.
```

ee7 replied:
Fixed, along with the same typo elsewhere.

ErikSchierboom commented on src/info/info.nim:
```diff
@@ -0,0 +1,72 @@
+import std/[algorithm, json, os, sets, terminal]
+import ".."/cli
+
+proc parseTrackConfig(trackDir: string): JsonNode =
```
We will need this in other places too (e.g. for linting), so it might be nice to have it in a shared module. We can do that later of course.

ee7 replied:
Now going via `jsony`. I might want to move the types out of `lint/track_config` at some point.

ErikSchierboom reviewed (approved):
Looks good. I'm not entirely sure about the command structure we should be using, which is why I've commented on https://github.com/exercism/configlet/issues/180. I'm fine with merging this on the condition that we might change the actual arguments later on depending on https://github.com/exercism/configlet/issues/180.

ee7 commented:
This PR caught my eye again for no particular reason, so I've:
- rebased it on `main`
- updated it to use `jsony`
- added the printing of unimplemented exercises

There wasn't a good reason that I didn't merge this PR - I guess I wanted to add a few more easy features first, but then other things were higher priority. So I guess I'll just merge this soon-ish.

As Erik says, we can think about CLI options for the `info` command later.
